AsyncDisplayKit/ASImageNode.mm

```diff
@@ -111,7 +111,9 @@ - (instancetype)initWithViewBlock:(ASDisplayNodeViewBlock)viewBlock
 - (CGSize)calculateSizeThatFits:(CGSize)constrainedSize
 {
   ASDN::MutexLocker l(_imageLock);
-  if (_image)
+  if (CGSizeEqualToSize(self.preferredFrameSize, CGSizeZero) == NO)
+    return CGSizeMake(MIN(constrainedSize.width, self.preferredFrameSize.width), MIN(constrainedSize.height, self.preferredFrameSize.height));
```

Comment:
haha, I can be annoying sometimes and you are very patient!

Thanks for this very interesting question. Short answer: I believe you have to set the node's frame in -layout method of its parent node. Long answer: With the new layout scheme, there are different ways you can use to force a size on an image node:
- Use ASRatioLayoutSpec and flexBasis. Example [here](https://github.com/facebook/AsyncDisplayKit/blob/master/examples/Kittens/Sample/KittenNode.mm#L138).
- Use ASStaticLayoutSpec.
- Manual layout: reserve enough space for the image node in -calculateSizeThatFits and set its frame in -layout. Example [here](https://github.com/facebook/AsyncDisplayKit/blob/master/examples/Kittens/Sample/KittenNode.mm#L180).

I cannot think of a use case in which the image node has different sizes based on its image's size. Even if there such use case, it's not a common one and should be handled by the user - custom image node. From the framework point of view, I think returning preferredFrameSize should be the default behaviour.